lib/PuppeteerSharp.Tests/InputTests/InputTests.cs

```diff
@@ -656,5 +656,96 @@ public async Task ShouldTypeEmojiIntoAniframe()
                 "👹 Tokyo street Japan \uD83C\uDDEF\uD83C\uDDF5",
                 await frame.QuerySelectorAsync("textarea").EvaluateFunctionAsync<string>("t => t.value"));
         }
+
+        [Fact]
+        public async Task ShouldScrollAccordingToMouseWheel()
+        {
+            var waitForFunctionOptions = new WaitForFunctionOptions
+            {
+                PollingInterval = 50,
+                Timeout = 1000
+            };
+
+            async Task WaitForScrollPoint(DomPointInternal point)
+                => await Page.WaitForFunctionAsync(
+                    @"(x, y) => document.body.scrollLeft === x && document.body.scrollTop === y",
+                    waitForFunctionOptions,
+                    point.X,
+                    point.Y);
+
+            await Page.GoToAsync(
+                $@"{TestConstants.ServerUrl}/longText.html",
+                WaitUntilNavigation.Networkidle0);
+
+            var expectedWheelEvents = new[]
+            {
+                new WheelEventInternal(0, 500),
+                new WheelEventInternal(0, -200),
+                new WheelEventInternal(300, 0),
+                new WheelEventInternal(-150, 0)
+            };
+
+            var expectedScrollPoint = new DomPointInternal
+            {
+                X = 0,
+                Y = 0
+            };
+
+            await WaitForScrollPoint(expectedScrollPoint);
+
+            foreach (var @event in expectedWheelEvents)
+            {
+                await Page.Mouse.WheelAsync(@event.DeltaX, @event.DeltaY);
+
+                expectedScrollPoint.Scroll(@event.DeltaX, @event.DeltaY);
+                await WaitForScrollPoint(expectedScrollPoint);
+            }
+        }
+
+        internal struct WheelEventInternal
+        {
+            public WheelEventInternal(decimal deltaX, decimal deltaY)
+            {
+                DeltaX = deltaX;
+                DeltaY = deltaY;
+            }
+
+            public decimal DeltaX
```

Comment:
Let's inline this props
```cs
public decimal DeltaX { get; set; }
```

Reply:
You mean just the code style - put accessors on the same line as property, correct?

Reply:
Exactly :)

Reply:
Done :)